proto/buf/alpha/registry/v1alpha1/doc.proto

```diff
@@ -377,6 +387,40 @@ message FieldOptions {
   int32 jstype = 4;
 }
 
+// Option provides information for a descriptor option that can be show in documentations.
+message Option {
+  // The name of the option.
+  string name = 1;
+  // The value of an option.
+  LiteralValue value = 2;
+}
+
+// MessageLiteral represents an option value for a type that is a message.
+message MessageLiteral {
+  map<string, LiteralValue> fields = 1;
```

Comment:
You may want a note here, similar to the comment above about option names, about the convention of the key. Something like:
```suggestion
  // Map of field names to value. Extension field names will be in the form:
  //     [fully.qualified.extension.Name]
  map<string, LiteralValue> fields = 1;
```

Reply:
I just realized that this will end up with no information about how to order the fields when rendering other than their name. It may be sufficient to simply sort the options by name (like with normal options first and custom options after). But it might be even better to sort them by their tag number, which would require a slightly different representation here.

Also, if we wanted more structural information about which fields were extensions and which weren't (like a separate `bool` field, like the comment suggests above), then you'd need to use something other than a map here. Maybe instead something like so:
```proto
message MessageLiteral {
  repeated FieldLiteral fields = 1;
}

message FieldLiteral {
  // When is_extension is true, name is the fully-qualified name of the
  // extension. Otherwise, it is the simple name of the field.
  string name = 1;
  int32 tag = 2;
  bool is_extension = 3;
  LiteralValue value = 4;
}
```
And at that point, it looks like `Option` (or `DescriptorOption`) maybe should just be `FieldLiteral`. So then the fields added to the various element types above would look like so:
```proto
  repeated FieldLiteral options = 10;
```

Reply:
Replaced `DescriptiorOption` with `FieldLiteral`